src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstiertodeliusupdate/client/HmppsTierApiClient.kt

```diff
@@ -22,7 +22,7 @@ class HmppsTierApiClient(@Qualifier("hmppsTierWebClientAppScope") private val we
     .uri("/crn/$crn/tier/$calculationId")
     .retrieve()
     .bodyToMono(TierDto::class.java)
-    .block()?.tierScore ?: throw EntityNotFoundException("No Tier record $calculationId found for $crn")
+    .block()!!.tierScore
```

Comment:
in the case of a 404, an error is thrown before we get this far. So this would only catch a 2xx response with no tier score. 

An alternative approach would be to handle the 404 as done here

https://github.com/ministryofjustice/hmpps-tier/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/AssessmentApiClient.kt#L35

Reply:
Or pass the nullable object out of here, but then we are introducing complexity that in reality doesn't exist.
There should be a better way really